internal/git/gitsync_processor.go

```diff
@@ -157,8 +185,228 @@ func watchRepo(ctx context.Context, r *git.Repository, gitSync *v1.GitSync, rest
 			return err
 		}
 	} else {
-		// TODO: monitoring with intervals
-		logger.Debug("monitoring with intervals")
+		ticker := time.NewTicker(timeInterval * time.Minute)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				patchedResources, recentHash, err := CheckForRepoUpdates(ctx, r, repo, &gitSync.Status, namespace)
```

Comment:
actually, what if there are no updates? It seems we need to return whether or not there are actually updates from here, right?

Reply:
Or we should skip the update if the commit status is the same, prefer to distinguish as early as possible.

Reply:
recentHash would be an empty string if there is no update in the repository ,so put a check on its length

Reply:
Could do it that way but I think it would be a little more readable to return a boolean for it personally

Reply:
I see what you added now. I was also thinking we should probably not call `updateCommitStatus()` if nothing changed

Reply:
In ```
commitStatus := v1.CommitStatus{
		Hash:     hash,
		Synced:   true,
		SyncTime: metav1.NewTime(time.Now()),
	}
``` we have SyncTime too it also wont be updated if there are no changes in the repo ,is that ok for us ? I mean does `SyncTime` have any major significance ?

Reply:
it's the last timestamp when changes were incorporated